docbrown/db/src/view_api/internal.rs

```diff
@@ -132,38 +132,46 @@ pub trait GraphViewInternalOps {
         d: Direction,
     ) -> Box<dyn Iterator<Item = u64> + Send>;
 
-    fn vertex_prop_vec(&self, v: VertexRef, name: String) -> Vec<(i64, Prop)>;
+    fn static_vertex_prop(&self, v: VertexRef, name: String) -> Option<Prop>;
 
-    fn vertex_prop_vec_window(
+    fn static_vertex_prop_keys(&self, v: VertexRef) -> Vec<String>;
+
+    fn temp_vertex_prop_vec(&self, v: VertexRef, name: String) -> Vec<(i64, Prop)>;
+
+    fn temp_vertex_prop_vec_window(
         &self,
         v: VertexRef,
         name: String,
         t_start: i64,
         t_end: i64,
     ) -> Vec<(i64, Prop)>;
 
-    fn vertex_props(&self, v: VertexRef) -> HashMap<String, Vec<(i64, Prop)>>;
+    fn temp_vertex_props(&self, v: VertexRef) -> HashMap<String, Vec<(i64, Prop)>>;
 
-    fn vertex_props_window(
+    fn temp_vertex_props_window(
         &self,
         v: VertexRef,
         t_start: i64,
         t_end: i64,
     ) -> HashMap<String, Vec<(i64, Prop)>>;
 
-    fn edge_props_vec(&self, e: EdgeRef, name: String) -> Vec<(i64, Prop)>;
+    fn static_edge_prop(&self, e: EdgeRef, name: String) -> Option<Prop>;
+
+    fn static_edge_prop_keys(&self, e: EdgeRef) -> Vec<String>;
+
+    fn temp_edge_props_vec(&self, e: EdgeRef, name: String) -> Vec<(i64, Prop)>;
 
-    fn edge_props_vec_window(
+    fn temp_edge_props_vec_window(
```

Comment:
perhaps change all `temp` to `temporal` as temp sounds more like temporary

Reply:
done